src/core/Akka.Remote/Transport/AkkaPduCodec.cs

```diff
@@ -228,14 +228,19 @@ protected AkkaPduCodec(ActorSystem system)
         /// <returns>TBD</returns>
         public virtual ByteString EncodePdu(IAkkaPdu pdu)
         {
-            ByteString finalBytes = null;
-            pdu.Match()
```

Comment:
Get rid of the old `.Match` extension method, which allocates. Replace with a C# 7 `switch` statement instead. Does not allocate.